### PR TITLE
refactor(cli): avoids big code blocks

### PR DESCRIPTION
**Pull request by minsk-dev**
Closes #<issue>

## ✅ Checklist

- [X] I have followed every step in the [contributing guide](https://github.com/t3-oss/create-t3-app/blob/main/CONTRIBUTING.md) (updated 2022-08-15).
- [X] The PR title follows the convention we established [conventional-commit](https://www.conventionalcommits.org/en/v1.0.0/)
- [X] I performed a functional test on my final commit

---

## Changelog

Moves some big blocks of code to functions, so the main function can be clearer/easier to read.

💯

### CODE REVIEW

File: cli/src/cli/index.ts [hunk 1]
```diff
@@ -185,76 +183,12 @@ export const runCli = async () => {
...
-
-        if (runInstall) {
-          logger.success("Alright. We'll install the dependencies for you!");
-        } else {
-          cliResults.flags.noInstall = true;
-          logger.info(
-            `No worries. You can run '${pkgManager} install' later to install the dependencies.`,
-          );
-        }
+        await askLanguage();
```
Comment (juliusmarminge) [inline cli/src/cli/index.ts:187]: I'd rename these to `promtX`

File: cli/src/cli/index.ts [hunk 2]
```diff
@@ -185,76 +183,12 @@ export const runCli = async () => {
...
+        cliResults.packages = await askPackages();
+        if (!cliResults.flags.noGit) cliResults.flags.noGit = !(await askGit());
+        if (!cliResults.flags.noInstall)
+          cliResults.flags.noInstall = !(await askInstall());
```
Comment (juliusmarminge) [inline cli/src/cli/index.ts:191]: kind of hard to read when some break lines and some don't. In these cases it's better to just force it I think:

```ts
if (!cliResults.flags.noGit) {
  cliResults.flags.noGit = !(await promptGit());
}
if () {
  ...
}
```

**Review by juliusmarminge — COMMENTED** [review]
I think it's a good change.

Also seems like you "missed" the name-part.

File: cli/src/cli/index.ts [hunk 3]
```diff
@@ -273,3 +207,81 @@ export const runCli = async () => {
...
+  return git;
+};
+
+const askInstall = async (): Promise<boolean> => {
+  const packageManager = getUserPkgManager();
+
+  const { install } = await inquirer.prompt<{ install: boolean }>({
+    name: "install",
+    type: "confirm",
+    message: `Would you like us to run ${packageManager} install?`,
```
Comment (juliusmarminge) [inline cli/src/cli/index.ts:274]:
```suggestion
    message: `Would you like us to run '${packageManager} install'?`,
```

**Comment by juliusmarminge** [comment]
> Also seems like you "missed" the name-part.

Bring that out to a separate function too for coherency

**Comment by minsk-dev** [comment]
Oh that's what you meant. Sure

**Review by juliusmarminge — APPROVED** [review]
Great 👍

File: cli/src/cli/index.ts [hunk 4]
```diff
@@ -166,94 +166,21 @@ export const runCli = async () => {
...
-          type: "confirm",
-          message: "Initialize a new git repository?",
-          default: true,
-        });
-        if (git) {
-          logger.success("Nice one! Initializing repository!");
-        } else {
-          cliResults.flags.noGit = true;
-          logger.info("Sounds good! You can come back and run git init later.");
+        cliResults.appName = await promptAppName();
```
Comment (juliusmarminge) [inline cli/src/cli/index.ts:175]: Missing ending bracket here I think? On mobile though so may be wrong?

Comment (minsk-dev) [reply cli/src/cli/index.ts:175]: It makes more sense if you try seeing only the last commit

Comment (c-ehrlich) [reply cli/src/cli/index.ts:175]: I'm also pretty sure that the logic is incorrect on this. Try this:
```
cd cli
pnpm build
pnpm start someAppName
```

Comment (juliusmarminge) [reply cli/src/cli/index.ts:175]: You are running every prompt conditionally depending on if the name was provided as option

Comment (minsk-dev) [reply cli/src/cli/index.ts:175]: @c-ehrlich You are right. I did this while doing my own project, so I lost some context. Fixed now

**Review by juliusmarminge — APPROVED** [review]
Looks correct. Since this logic isn't caught by CI I suggest we do some manual tests to be sure before merging.

**Comment by c-ehrlich** [comment]
behaves correctly both with and without name provided to the cli

image

**Comment by minsk-dev** [comment]
I retested being pushing too

**Comment by juliusmarminge** [comment]
Cool I'll merge, we can test @beta afterwards

**Pull request merged**